src/group-constructors.js

```diff
@@ -37,6 +41,55 @@
   };
 
   constructor.prototype = {
+    _isAncestor: function(effect) {
+      var a = this;
+      while (a !== null) {
+        if (a == effect)
+          return true;
+        a = a._parent;
+      }
+      return false;
+    },
+    _rebuild: function() {
+      // Re-calculate durations for ancestors with specified duration 'auto'.
+      var node = this;
+      while (node) {
+        if (node.timing.duration === 'auto') {
+          node._timing.duration = node.activeDuration;
+        }
+        node = node._parent;
+      }
+      if (this.animation) {
```

Comment:
I guess this is the effect so it probably is animation, and should be _animation?